src/init.cpp

```diff
@@ -1558,7 +1558,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
 
         node::ChainstateLoadOptions options;
         options.mempool = Assert(node.mempool.get());
-        options.reindex = chainman.m_blockman.m_reindexing;
+        options.reindex = blockman_opts.reindex;
```

Comment:
nit: How about taking the opportunity to nail down the mutability of `ChainstateLoadOptions` for improved readability?
```suggestion
        const node::ChainstateLoadOptions options{
            .mempool = Assert(node.mempool.get()),
            .reindex = blockman_opts.reindex,
            .reindex_chainstate = fReindexChainState,
            .prune = chainman.m_blockman.IsPruneMode(),
            .require_full_verification = args.IsArgSet("-checkblocks") || args.IsArgSet("-checklevel"),
            .check_blocks = args.GetIntArg("-checkblocks", DEFAULT_CHECKBLOCKS),
            .check_level = args.GetIntArg("-checklevel", DEFAULT_CHECKLEVEL),
            .coins_error_cb = [] {
                uiInterface.ThreadSafeMessageBox(
                    _("Error reading from database, shutting down."),
                    "", CClientUIInterface::MSG_ERROR);
            },
        };
```